build/Jamfile.v2

```diff
@@ -214,7 +208,7 @@ rule select-arch-specific-sources ( properties * )
 {
     local result ;
 
-    if x86 in [ log-arch-config.deduce-architecture $(properties) ]
+    if <architecture>x86 in $(properties)
```

Comment:
Is this correct? This must work as expected even if `<architecture>` is not specified by user. Same with the similar change re. `<address-model>` below.